File: src/main/java/dev/jarand/authprotectedrequests/cookie/CookieServiceMock.java
package dev.jarand.authprotectedrequests.cookie;

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.stereotype.Service;

import javax.servlet.http.Cookie;

@Service
@ConditionalOnProperty(name = "authentication.mock.enabled", havingValue = "true")
public class CookieServiceMock implements CookieService {

    @Override
    public Cookie createAccessTokenCookie(String accessToken) {
        throw new IllegalStateException("Not implemented");
    }
}

File: src/main/kotlin/dev/jarand/authprotectedrequests/cookie/CookieServiceMock.kt
package dev.jarand.authprotectedrequests.cookie

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.stereotype.Service
import javax.servlet.http.Cookie

@Service
@ConditionalOnProperty(name = ["authentication.mock.enabled"], havingValue = "true")
class CookieServiceMock : CookieService {

    override fun createAccessTokenCookie(accessToken: String): Cookie {
        throw IllegalStateException("Not implemented")
    }
}
